src/proposer/proposer_rpc.cpp

```diff
@@ -167,6 +170,125 @@ class ProposerRPCImpl : public ProposerRPC {
     obj.pushKV("stakeable_coins", arr);
     return obj;
   }
+
+  UniValue ProposeBlocks(CWallet &wallet,
+                         const boost::optional<CScript> &coinbase_script,
+                         const int num_generate) const {
+
+    proposer::Proposer *proposer = GetComponent<proposer::Proposer>();
+
+    UniValue block_hashes(UniValue::VARR);
+
+    // To pick up to date coins for staking we need to make sure that the wallet is synced to the current chain.
+    wallet.BlockUntilSyncedToCurrentChain();
+
+    esperanza::WalletExtension &wallet_ext = wallet.GetWalletExtension();
+
+    for (int i = 0; i < num_generate; ++i) {
```

Comment:
An unsigned int, or an `std::size_t`, seems a bit more applicable to me here.

Reply:
If I am not mistaken, our rpc api allows only signed ints, so it does make sense to be `int` since num_generate is int

Reply:
I see. So sad. Fine with me.